File: vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

bool isValidInput(string input) 
{
	for(int i=0; i<strlen(input); i++) 
	{
		char ch = input[i];
		if (!((ch >='a' && ch <='z')||(ch>='A' && ch <='Z'))) 
		{
			return false;
		}
	}
	return true;
}


int main(int argc, string argv[])
{
	if (argc != 2 || !isValidInput(argv[1])) {
		printf("Usage: ./vigenere k\n");
		return 1;
	}
	int h = 0;
	string key = argv[1];
	printf("plaintext: ");
	string text = get_string();
	printf("ciphertext: ");
	for(int i=0, n=strlen(text); i<n; i++)
	{
		int j= h%strlen(key);
		int k = toupper(key[j])-'A';
		char ch = text[i];
		if (ch >= 'A' && ch <= 'Z') 
		{
			char index = (ch - 'A' + k) % 26;
			printf("%c", (char)(index + 'A'));
			h++;
		} 
		else if (ch >= 'a' && ch <= 'z') 
		{
			char index = (ch - 'a' + k) % 26;
			printf("%c", (char)(index + 'a'));
			h++;
		} 
		else {
			printf("%c", ch);
		}
	}
	printf("\n");
	return 0;
}

File: vigenere.py
import cs50
import sys

def main():
    if len(sys.argv) != 2 or not(sys.argv[1].isalpha()):
        print("Usage: ./vigenere k")
        return (1)
    else:
        key = sys.argv[1].upper()
        print("plaintext: ", end="")
        text = cs50.get_string()
        print("ciphertext: ", end="")
        i = 0
        for c in text:
            k = ord(key[i%len(key)]) - 65
            if c.isalpha() and c.isupper():
                position = (ord(c) - 65 + k) % 26
                print(chr(position + 65), end="")
                i += 1
            elif c.isalpha and c.islower():
                position = (ord(c) - 97 + k) % 26
                print(chr(position + 97), end="")
                i += 1
            else:
                print(c, end="")
        print()
        return (0)   
    
    
if __name__ == "__main__":
    main()